### PR TITLE
Draft document classifier

### PR DESCRIPTION
**Pull request by vitawasalreadytaken**
I've placed two somewhat unrelated things into this branch because they both depend on #6 and on some research-y code from our old, exploratory repository.

- `research/consultation_topics/label_exploration.ipynb` takes a look at the consultation topic labels we have.
- `research/document_types/VM_draft_classifier.ipynb` is a bare-bone document type classifier, on Fedlex data only, making "is DRAFT/is not a DRAFT" predictions only.

We can split these up again if this branch stops making sense.

Handing this over to you now, @orieger 🙂

### CODE REVIEW

File: research/document_types/draft_classification.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    with embeddings.use_cache(
+            embedding_model,
+            tqdm=tqdm,
+            cache_directory=cache_directory,
+    ) as get_embeddings:
+        embeddings_doc_content_plain = get_embeddings(tokens.tolist())
+        print(embeddings_doc_content_plain.shape)
+    return embeddings_doc_content_plain
+
+def save_model(model_file_name):
```
Comment (vitawasalreadytaken) [inline research/document_types/draft_classification.py:62]: This is similar - let's pass `classifier` in as an argument.

File: research/document_types/draft_classification.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from tqdm import tqdm
+import toml
+
+from research.lib import data_access, embeddings
+
+REPOSITORY_ROOT = (pathlib.Path().cwd() / ".." / "..").resolve()
+sys.path.append(str(REPOSITORY_ROOT))
```
Comment (vitawasalreadytaken) [inline research/document_types/draft_classification.py:25]: Is this necessary? Since the imports above (`from research.lib ...`) apparently work?

File: research/document_types/draft_classification.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+dotenv.load_dotenv()
```
Comment (vitawasalreadytaken) [inline research/document_types/draft_classification.py:27]: Side effects such as this should be in the `if __name__ == "__main__":` block, otherwise they're triggered even when this module is imported by something else.

File: research/document_types/draft_classification.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+with open('draft_classification.toml', 'r') as f:
+    config = toml.load(f)
+
+os.environ["MLFLOW_TRACKING_URI"] = config["tracking"]["tracking_uri"]
```
Comment (vitawasalreadytaken) [inline research/document_types/draft_classification.py:32]: This is also a side effect that should not be triggered at import time. In addition, we can use https://mlflow.org/docs/1.22.0/python_api/mlflow.html#mlflow.set_tracking_uri instead of manipulating environment variables. (I know it was like that in the original notebook but we can improve things in this new script 🙂)

File: research/document_types/draft_classification.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+from research.lib import data_access, embeddings
+
+REPOSITORY_ROOT = (pathlib.Path().cwd() / ".." / "..").resolve()
+sys.path.append(str(REPOSITORY_ROOT))
+
+dotenv.load_dotenv()
+
+with open('draft_classification.toml', 'r') as f:
+    config = toml.load(f)
```
Comment (vitawasalreadytaken) [inline research/document_types/draft_classification.py:30]:
```suggestion
config = toml.load("draft_classification.toml")
```

However, it's also a side effect that would preferably happen in `if __name__ == "__main__"`...

File: research/document_types/draft_classification.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        only_languages=LANGUAGES,
+        only_doc_types=DOC_TYPES,
+        starting_year=FROM_YEAR,
+        mlflow=mlflow,
+    )
+
+    ### Preprocessing ###
+    df_input = remove_rows_with_missing_text(df_input)
+    # set target variable
+    df_input.loc[:, "is_draft"] = (df_input.loc[:, "document_type"] == "DRAFT").astype(bool)
```
Comment (vitawasalreadytaken) [inline research/document_types/draft_classification.py:104]: Oddly, I'm getting this warning on this line. I don't know why.

```
SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead
```

File: research/document_types/draft_classification.py (hunk 2 above)
Comment (vitawasalreadytaken) [reply research/document_types/draft_classification.py:25]: Note: I had to run the script like this: 

```
cd research/document_types
PYTHONPATH=../../ uv run draft_classification.py
``` 

so that a) the `research.lib` imports on line 22 work and b) the relative path to `draft_classification.toml` works.

**Comment by vitawasalreadytaken** [comment]
Closing in favour of #15
